automation/src/main/resources/puppet/manifests/dawg.pp

```diff
@@ -0,0 +1,97 @@
+class dawg::dawg {
+    class { 'java':
+        package => hiera('java.package', 'java-1.7.0-openjdk'),
+    }
+
+    $tomcatSrc = hiera('tomcat.src', 'http://archive.apache.org/dist/tomcat/tomcat-7/v7.0.73/bin/apache-tomcat-7.0.73.tar.gz')
+    tomcat::install { '/opt/tomcat':
+        source_url => $tomcatSrc,
+    }
+
+    file { '/etc/init.d/tomcat':
+        content => hiera('tomcat.init'),
+        mode => '0755',
+        require => Tomcat::Install['/opt/tomcat'],
+    }
+
+    exec {"genunit-tomcat":
+        path => '/sbin:/bin:/usr/sbin:/usr/bin',
+        onlyif => 'test -f /bin/systemctl',
+        command => 'systemctl daemon-reload',
+        require => File["/etc/init.d/tomcat"],
+    }
+
+    # The tomcat puppet module has a resource that does this, but it sucks
+    service { 'tomcat':
+        ensure => running,
+        require => [File['/etc/init.d/tomcat'],Exec['genunit-tomcat']],
+    }
+
+    $nexurl = hiera('nexus.url')
+
+	class {'nexus':
+	   url => $nexurl
+    }
+
+    $version = hiera('dawg.version')
+    $repo = hiera('nexus.repo')
+    $groupId = "com.comcast.video.dawg"
+
+    $dsArtifactId = "dawg-show"
+    nexus::artifact { $dsArtifactId:
+            gav => "${groupId}:${dsArtifactId}:${version}",
+            repository => $repo,
+            output => "/tmp/${dsArtifactId}-${version}.war",
+            packaging => 'war',
+            ensure => 'present',
+   }
+
+    $dhArtifactId = "dawg-house"
+    nexus::artifact { $dhArtifactId:
+            gav => "${groupId}:${dhArtifactId}:${version}",
+            repository => $repo,
+            output => "/tmp/${dhArtifactId}-${version}.war",
+            packaging => 'war',
+            ensure => 'present',
+   }
+
+    $dpArtifactId = "dawg-pound"
+    nexus::artifact { $dpArtifactId:
+            gav => "${groupId}:${dpArtifactId}:${version}",
+            repository => $repo,
+            output => "/tmp/${dpArtifactId}-${version}.war",
+            packaging => 'war',
+            ensure => 'present',
+   }
+
+   file { '/etc/dawg':
+        ensure => 'directory',
+        owner => 'tomcat',
+   }
+
+   config_file::config_file { 'dawg.ini':
```

Comment:
it will only work internally for now. The puppet module gets installed on the machine and then it can be referenced. I suppose I can just use an erb template for now. But its pretty likely this automation won't be needed by others very soon.